subsys/bluetooth/audio/tmap.c

```diff
@@ -22,63 +22,97 @@
 
 LOG_MODULE_REGISTER(bt_tmap, CONFIG_BT_TMAP_LOG_LEVEL);
 
+/* Hex value if all TMAP role bits are set */
+#define TMAP_ALL_ROLES		0x3F
+
 static uint16_t tmap_role;
 static struct bt_tmap_cb *cb;
 static bool tmas_found;
 
-static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
-static struct bt_gatt_discover_params discover_params;
-static struct bt_gatt_read_params read_params;
+static struct bt_uuid_16 uuid[CONFIG_BT_MAX_CONN] = {BT_UUID_INIT_16(0)};
+static struct bt_gatt_discover_params discover_params[CONFIG_BT_MAX_CONN];
+static struct bt_gatt_read_params read_params[CONFIG_BT_MAX_CONN];
 
 uint8_t tmap_char_read(struct bt_conn *conn, uint8_t err,
 					struct bt_gatt_read_params *params,
 					const void *data, uint16_t length)
 {
-	enum bt_tmap_role peer_role = sys_get_le16(data);
+	/* Check read request result */
+	if (err != BT_ATT_ERR_SUCCESS) {
+		if (cb->discovery_complete) {
+			cb->discovery_complete(0, conn, err);
+		}
 
-	if (cb->tmap_discovery_complete) {
-		cb->tmap_discovery_complete(peer_role, conn, err);
+		return BT_GATT_ITER_STOP;
 	}
+
+	/* Check data length */
+	if (length != sizeof(uint16_t)) {
+		if (cb->discovery_complete) {
+			cb->discovery_complete(0, conn, BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+		}
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	/* Extract the TMAP role of the peer and inform application fo the value found */
+	uint16_t peer_role = sys_get_le16(data);
```

Comment:
Declare `peer_role` at top of scope and do

```suggestion
	if (length != sizeof(peer_role)) {
		if (cb->discovery_complete) {
			cb->discovery_complete(0, conn, BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
		}

		return BT_GATT_ITER_STOP;
	}

	/* Extract the TMAP role of the peer and inform application fo the value found */
	peer_role = sys_get_le16(data);
```